cql/ManageSpecialPopulation.cql

```diff
@@ -256,16 +268,88 @@ define RecommendationForPatientsYoungerThan25:
           'A diagnostic excisional procedure is recommended in patients when the squamocolumnar junction or the upper limit of all lesions are not fully visualized.'
         }
       }    
-    when ( // Heading 3.2
-      AgeInYears() >= 21 and AgeInYears() < 25 and
+
+    when ( // K.1 Heading 3 case 2
```

Comment:
The recommendation for CIN1 preceded by HSIL is to get a colpo and cytology in 1 year. The recommendation for CIN1 preceded by ASC-H is to get a cytology in 1 year. If a patient followed the guidelines, in order to have one of these high grade cytology results persist for two years, they would have had to have had the original CIN1 preceded by ASC-H/HSIL, followed by another cytology or cytology+colpo at the 1 year mark, followed by yet a 3rd cytology+colpo at the 2 year mark. Therefore, I believe labelling the original CIN1 preceded by HSIL/ASCH here as the "SecondMostRecent" breaks down. 

The test you added - `Young33HSILCytoCin1Persists.yml` - is not representative of what somebody following the guidelines would do, because the synthetic patient waited over 2 years to come back for a follow up, instead of coming back in 1 year.

Reply:
One idea of how to frame this logic would be to look for the most recent cytology to be ASC-H or HSIL, and then if that's true, check if there is ANY CIN1 preceded by ASC-H or HSIL cytology between 2-3 years before that most recent cytology test.

Reply:
I believe there is overlap in the L2 logic for K.1.3.2(b) and K.1.3.3. We should discuss how to resolve.

Reply:
You're right, there is. That's the reason we evaluate K.1.3.3 before K.1.3.2(b). Because the recommendation in K.1.3.3 is more specific. But that's an entirely different issue.

What I was describing above is that K.1.3.3 is a recommendation for "If a high-grade cytologic abnormality (HSIL, ASC-H) without histologic HSIL persists for 2 years". It's not saying that if your most recent and second most recent test were HSIL/ASC-H, than you should get a diagnostic excisional procedure. It's saying that 2-3 years ago they had HSIL/ASC-H, and now they still do. But those two tests don't have to be subsequent, there can, and should, be other tests in between. You've inserted "MostRecent" and "SecondMostRecent" logic here that doesn't align with recommendation K.1.3.3 in the L2.

Reply:
Removed the use of SecondMostRecent here. Changed logic to check if there is ANY CIN1 preceded by ASC-H or HSIL cytology between 2-3 years before that most recent cytology test AND check there is no treatment after the most recent cytology.